package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua

```diff
@@ -0,0 +1,79 @@
+local uci = require("simple-uci").cursor()
+
+local util = require "gluon.util"
+
+local M = {}
+
+function M.enabled()
+	return uci:get_bool("gluon", "mesh_vpn", "enabled")
+end
+
+function M.enable(val)
+	return uci:set("gluon", "mesh_vpn", "enabled", val)
+end
+
+function M.get_interface()
+	return 'mesh-vpn'
+end
+
+function M.get_method(name)
+	return require("gluon.mesh-vpn.method." .. name)
+end
+
+function M.get_method_names()
+	local out = {}
+
+	for _, v in ipairs(util.glob("/lib/gluon/mesh-vpn/method/*")) do
+		table.insert(out, v:match('([^/]+)$'))
+	end
+
+	return out
+end
+
+function M.get_active_method_name()
+	for _, name in ipairs(M.get_method_names()) do
+		local method = M.get_method(name)
+
+		if method.available() then
+			return name
+		end
+	end
+
+	return nil
+end
```

Comment:
As `get_active_method_name()` needs to load the method modules anyways, it would be nicer to merge it into `get_active_method()` (which can then have two return values, the name and the implementation).